src/com/sap/piper/variablesubstitution/Logger.groovy

```diff
@@ -0,0 +1,37 @@
+package com.sap.piper.variablesubstitution
+
+/**
+ * Very simple logger class that can be instantiated to
```

Comment:
See the reply on introducing the `Logger` above.

The current code uses a simplistic Logger encapsulating the code that checks whether or not a message should be logged based on the configuration. 
For that to work, the Logger needs the configuration which is not present (according to the pattern of `handlePipelineStepErrors` and `ConfigurationHelper`) until the `call(...)` method is executed.

An alternative would be to put ugly `if(config.verbose)` statements in all the places where now the simple Logger is called. And quite frankly, I don't think that is a better solution - although I have seen it in other coding as well.

Reply:
I'm not against a logger, maybe that is something which is really missing.
But then lets introduce a logger in a dedicated PR, merge that PR and continue with this PR.
IMO it is not a cool approach to introduce a logger as a side effect of an PR which focuses on variable substitution in yaml files.

Reply:
Thanks @marcusholl, but provided that I do not have the time to introduce a new Logger system now, what do you suggest I do with the current PR?
Park it? Remove the logger and litter the code with if-statements? 
Why will an iterative approach not do just as well? - Introduce a stripped down version of Logger which does what is currently most urgently needed (from my point of view) and replace that with a more general solution once we have it.

Reply:
I just found that: https://github.com/wcm-io-devops/jenkins-pipeline-library/blob/master/docs/logging.md

Is there a reason why we cannot use that in the Jenkins Library?

Reply:
In principle it would be possible to use that. As always there are pros and cons (_no I can start to outline the details, for the sake of shortness I will omit that_). Thanks for sharing, I will keep that in mind and also discuss with colleagues.